insights/parsers/smartctl.py

```diff
@@ -35,12 +35,13 @@ class SMARTctl(Parser):
 
     def __init__(self, context):
         super(SMARTctl, self).__init__(context)
+        self.data = {}
         filename_re = re.compile(r'smartctl_-a_\.dev\.(?P<device>\w+)$')
         match = filename_re.search(context.path)
-        self.data = {}
         if match:
             self.device = match.group('device')
-        # Else warn?  fail?
+        else:
+            self.device = ''
```

Comment:
@PaulWay any idea if this occurs?  Seems like it might be better to indicate this as `Unknown` or some specific value to make it more explicit.  Have you seen any examples of this in actual data?